blockchain/indexers/indexsubscriber.go

```diff
@@ -348,43 +354,85 @@ func (s *IndexSubscriber) CatchUp(ctx context.Context, db database.DB, queryer C
 	return nil
 }
 
-// Run relays index notifications to subscribed indexes.
+// handleSyncSubscribers updates index sync subscribers when a subscribed
+// indexer is fully synced.
 //
 // This should be run as a goroutine.
-func (s *IndexSubscriber) Run(ctx context.Context) {
+func (s *IndexSubscriber) handleSyncSubscribers(ctx context.Context) {
+	ticker := time.NewTicker(syncUpdateInterval)
+
 	for {
 		select {
-		case ntfn := <-s.c:
-			// Relay the index update to subscribed indexes.
+		case <-ctx.Done():
+			ticker.Stop()
```

Comment:
I'd prefer this be done via a defer right after the creation for consistency with every other use of ticket in the code base.